php/composer/Portfile

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        composer composer 2.5.8
+revision            0
+categories          php devel
+platforms           any
+license             MIT
+maintainers         {ego.team:yet @yugaego} openmaintainer
+
+description         a dependency manager for the PHP programming language
+
+long_description    Composer is {*}${description}. It helps you declare, \
+                    manage, and install dependencies of PHP projects.
+
+homepage            https://getcomposer.org
+
+github.tarball_from releases
+distname            ${name}
+extract.suffix      .phar
+
+revision            0
+checksums           rmd160  cbcc680e5206a5f016151ab15824685523538819 \
+                    sha256  f07934fad44f9048c0dc875a506cca31cc2794d6aebfc1867f3b1fbf48dce2c5 \
+                    size    2837394
+
+depends_run         bin:php:php82
+
+extract {
+    file copy ${distpath}/${distfiles} ${workpath}/${distname}
+}
+
+use_configure no
+
+build {}
+
+destroot {
+    xinstall -m 0755 ${workpath}/${distname} ${destroot}${prefix}/bin/
```

Comment:
This will install /opt/local/bin/composer.phar. I glanced at the [installation instructions](https://getcomposer.org/doc/00-intro.md#globally) and they suggested installing it as `composer`, not `composer.phar`.

Looking at [repology](https://repology.org/project/php:composer/versions), most collections seem to offer only one composer package (possibly with a mechanism like MacPorts variants to choose the php version), but Void Linux offers several packages, one for each php version (which we would model as subports). My instinct is that subports, rather than variants, are the way to go, in which case the executable name would have the php version in it somewhere, e.g. `composer83`. A composer_select port could then be introduced to help the user create a `composer` symlink pointing to whichever subport of composer they have installed (if it's important that it be reachable by typing `composer`).